### PR TITLE
Update more dependencies.

### PR DESCRIPTION
**Pull request by anuraaga**
@jkwatson I found the junit issue you mentioned, will sort it out separately

### CODE REVIEW

File: exporters/otlp/common/src/test/java/io/opentelemetry/exporter/otlp/internal/traces/SpanDataRandomizerRegistry.java [hunk 1]
```diff
@@ -1,361 +0,0 @@
-/*
```
Comment (anuraaga) [inline exporters/otlp/common/src/test/java/io/opentelemetry/exporter/otlp/internal/traces/SpanDataRandomizerRegistry.java:1]: Forgot to delete this class in the other PR

File: sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/MetricExporterConfiguration.java [hunk 2]
```diff
@@ -172,7 +172,7 @@ private static void configurePrometheusMetrics(
     } catch (IOException e) {
       throw new IllegalStateException("Failed to create Prometheus server", e);
     }
-    Runtime.getRuntime().addShutdownHook(new Thread(server::stop));
+    Runtime.getRuntime().addShutdownHook(new Thread(server::close));
```
Comment (jkwatson) [inline sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/MetricExporterConfiguration.java:175]: Is this the same thing, just renamed, or a different function?

**Review by jkwatson — APPROVED** [review]

File: sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/MetricExporterConfiguration.java (hunk 2 above)
Comment (anuraaga) [reply sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/MetricExporterConfiguration.java:175]: Yeah - previous one is deprecated and our build prevents using deprecated stuff.

Comment (anuraaga) [reply sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/MetricExporterConfiguration.java:175]: It does make me wonder how to release autoconfigure as stable with the dependency on this pre-1.0 Prometheus artifact which is known to break some times. We may need to go ahead and implement our own endpoint...

**Pull request merged**